code/Controllers/CMSMain.php

```diff
@@ -1038,56 +1038,93 @@ public function getBreadcrumbsBackLink()
             ->Link;
     }
 
+    /**
+     * Generate a breadcrumb based on the page currently being viewed.
+     * @param bool $unlinked Wheter to include links
+     * @return ArrayList<ArrayData>
+     */
     public function Breadcrumbs($unlinked = false)
     {
-        $items = new ArrayList();
-
-        if ($this->TreeIsFiltered()) {
-            $items->push(new ArrayData([
-                'Title' => CMSPagesController::menu_title(),
-                'Link' => ($unlinked) ? false : $this->LinkPages()
-            ]));
-            $items->push(new ArrayData([
-                'Title' => _t('SilverStripe\\CMS\\Controllers\\CMSMain.SEARCHRESULTS', 'Search results'),
-                'Link' => ($unlinked) ? false : $this->LinkPages()
-            ]));
+        $items = ArrayList::create();
 
-            $this->extend('updateBreadcrumbs', $items);
+        if (($this->getAction() !== 'index') && ($record = $this->currentPage())) {
+            // The page is being edited
+            $this->buildEditFormBreadcrumb($items, $record, $unlinked);
+        } else {
+            // All other scenarios include a "Pages" first crumb
+            $this->pushCrumb(
+                $items,
+                CMSPagesController::menu_title(),
+                $unlinked ? false : $this->LinkPages()
+            );
 
-            return $items;
+            if ($this->TreeIsFiltered()) {
+                // Showing search results
+                $this->pushCrumb(
+                    $items,
+                    _t(CMSMain::class . '.SEARCHRESULTS', 'Search results'),
+                    ($unlinked) ? false : $this->LinkPages()
+                );
+            } elseif ($parentID = $this->getRequest()->getVar('ParentID')) {
+                // We're navigating the listview. ParentID is the page whose
+                // children are currently displayed.
+                if ($page = SiteTree::get()->byID($parentID)) {
+                    $this->buildListViewBreadcrumb($items, $page);
+                }
+            }
         }
 
-        // Check if we are editing a page
-        /** @var SiteTree $record */
-        $record = $this->currentPage();
-        if (!$record) {
-            $items->push(new ArrayData([
-                'Title' => CMSPagesController::menu_title(),
-                'Link' => ($unlinked) ? false : $this->LinkPages()
-            ]));
+        $this->extend('updateBreadcrumbs', $items);
 
-            $this->extend('updateBreadcrumbs', $items);
+        return $items;
+    }
 
-            return $items;
-        }
+    /**
+     * Push the provided an extra breadcrumb crumb at the end of the provided List
+     */
+    private function pushCrumb(ArrayList $items, string $title, string|false $link): void
+    {
+        $items->push(ArrayData::create([
+            'Title' => $title,
+            'Link' => $link
+        ]));
+    }
```

Comment:
This extra method seems very unnecessary but not worth the ping pong to remove it, since it's private anyway.